go/enclave/events/subscription_manager.go

```diff
@@ -169,26 +169,30 @@ func (s *SubscriptionManager) encryptLogs(logsByID map[gethrpc.ID][]*types.Log)
 	return encryptedLogsByID, nil
 }
 
-// Extracts the (potential) user addresses from the topics. If there is no code associated with an address, it's a user
-// address.
+// Extracts the user addresses from the topics.
 func getUserAddrsFromLogTopics(log *types.Log, db *state.StateDB) []string {
 	var userAddrs []string //nolint:prealloc
 
 	for _, topic := range log.Topics {
+		topicHex := topic.Hex()
+		potentialAddr := gethcommon.HexToAddress(topicHex)
+
 		// Since addresses are 20 bytes long, while hashes are 32, only topics with 12 leading zero bytes can
 		// (potentially) be user addresses.
-		topicHex := topic.Hex()
 		if topicHex[2:len(zeroBytesHex)+2] != zeroBytesHex {
 			continue
 		}
 
-		// If there is code associated with an address, it is not a user address.
-		addr := gethcommon.HexToAddress(topicHex)
-		if db.GetCode(addr) != nil {
+		// We also assume that a potential address is *not* a user address if any of the following is true:
```

Comment:
it's a bit hard to follow being negative. 
I think it would e clearer if you appended in the if branch, and made the entire thing positive

Reply:
@tudor-malene Fixed. Can you take another look?